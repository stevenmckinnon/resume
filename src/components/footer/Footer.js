import React from 'react';

class Footer extends React.Component {
    render() {
        return (
            <footer>
                <div className="row">
                    <div className="twelve columns">
                        <ul className="social-links">
                            {this.props.socialMedia.twitter ? <li><a href={this.props.socialMedia.twitter}>
                                <i className="fa fa-twitter" /></a></li> : null}
                            {this.props.socialMedia.linkedIn ? <li><a href={this.props.socialMedia.linkedIn}>
                                <i className="fa fa-linkedin" /></a></li> : null}
                            {this.props.socialMedia.instagram ? <li><a href={this.props.socialMedia.instagram}>
                                <i className="fa fa-instagram" /></a></li> : null}
                            {this.props.socialMedia.photography ? <li><a href={this.props.socialMedia.photography} 
                                title={`${this.props.name} Photography`}><i className="fa fa-camera" /></a></li> : null}
                            {this.props.socialMedia.github ? <li><a href={this.props.socialMedia.github}>
                                <i className='fa fa-github'></i></a></li> : null}
                        </ul>
                        <ul className="copyright">
                            <li>Â© Copyright {new Date().getFullYear()} Steven McKinnon</li>
                        </ul>
                    </div>
                    <div id="go-top"><a className="smoothscroll" title="Back to Top" href="#home"><i className="icon-up-open" /></a></div>
                </div>
            </footer>
        );
    }
}

export default Footer;