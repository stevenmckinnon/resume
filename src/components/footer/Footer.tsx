import * as React from 'react';
import { Link } from "react-scroll";
import { SocialMedia } from '../../interfaces';

interface Props {
    name: string;
    socialMedia: SocialMedia;
}

const Footer: React.FC<Props> = ({ name, socialMedia }) => (
    <footer>
        <div className="row social">
            <div className="twelve columns">
                <ul className="social-links">
                    {socialMedia.twitter && <li><a href={socialMedia.twitter} target="_blank" rel="noopener noreferrer">
                        <i className="fab fa-twitter" /></a></li>}
                    {socialMedia.linkedIn && <li><a href={socialMedia.linkedIn} target="_blank" rel="noopener noreferrer">
                        <i className="fab fa-linkedin" /></a></li>}
                    {socialMedia.instagram && <li><a href={socialMedia.instagram} target="_blank" rel="noopener noreferrer">
                        <i className="fab fa-instagram" /></a></li>}
                    {socialMedia.photography && <li><a href={socialMedia.photography} target="_blank" rel="noopener noreferrer"
                        title={`${name} Photography`}><i className="fa fa-camera" /></a></li>}
                    {socialMedia.github && <li><a href={socialMedia.github} target="_blank" rel="noopener noreferrer">
                        <i className='fab fa-github'></i></a></li>}
                    {socialMedia.buyMeACoffee && <li><a href={socialMedia.buyMeACoffee} target="_blank" rel="noopener noreferrer">
                        <i className='fa fa-coffee'></i></a></li>}
                </ul>
                <ul className="copyright">
                    <li>Â© Copyright {new Date().getFullYear()} {name}</li>
                </ul>
            </div>
            <div id="go-top">
                <Link className="smoothscroll" title="Back to Top" to="home" smooth={true}>
                    <i className="fas fa-chevron-up" />
                </Link>
            </div>
        </div>
    </footer>
);

export default Footer;