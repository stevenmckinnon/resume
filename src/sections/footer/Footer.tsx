import * as React from "react"
import { Link } from "react-scroll"
import { type SocialMedia } from "../../interfaces"
import {
  Camera,
  ChevronUp,
  Coffee,
  Github,
  Instagram,
  Linkedin,
  Twitter,
} from "lucide-react"

interface Props {
  name: string
  socialMedia: SocialMedia
}

const Footer: React.FC<Props> = ({ name, socialMedia }) => (
  <footer className="bg-primary flex flex-col gap-3 justify-center items-center p-10">
    <ul className="flex gap-3">
      {socialMedia.twitter && (
        <li>
          <a
            href={socialMedia.twitter}
            target="_blank"
            rel="noopener noreferrer"
            className="text-xl text-white hover:text-teal-300 cursor-pointer"
          >
            <Twitter />
          </a>
        </li>
      )}
      {socialMedia.linkedIn && (
        <li>
          <a
            href={socialMedia.linkedIn}
            target="_blank"
            rel="noopener noreferrer"
            className="text-xl text-white hover:text-teal-300 cursor-pointer"
          >
            <Linkedin />
          </a>
        </li>
      )}
      {socialMedia.instagram && (
        <li>
          <a
            href={socialMedia.instagram}
            target="_blank"
            rel="noopener noreferrer"
            className="text-xl text-white hover:text-teal-300 cursor-pointer"
          >
            <Instagram />
          </a>
        </li>
      )}
      {socialMedia.photography && (
        <li>
          <a
            href={socialMedia.photography}
            target="_blank"
            rel="noopener noreferrer"
            title={`${name} Photography`}
            className="text-xl text-white hover:text-teal-300 cursor-pointer"
          >
            <Camera />
          </a>
        </li>
      )}
      {socialMedia.github && (
        <li>
          <a
            href={socialMedia.github}
            target="_blank"
            rel="noopener noreferrer"
            className="text-xl text-white hover:text-teal-300 cursor-pointer"
          >
            <Github />
          </a>
        </li>
      )}
      {socialMedia.buyMeACoffee && (
        <li>
          <a
            href={socialMedia.buyMeACoffee}
            target="_blank"
            rel="noopener noreferrer"
            className="text-xl text-white hover:text-teal-300 cursor-pointer"
          >
            <Coffee />
          </a>
        </li>
      )}
    </ul>
    <ul className="copyright">
      <li>
        <p className="text-white font-body">
          Â© Copyright {new Date().getFullYear()} {name}
        </p>
      </li>
    </ul>
    <Link
      className="text-2xl text-white hover:text-teal-300 cursor-pointer"
      title="Back to Top"
      to="home"
      smooth={true}
    >
      <ChevronUp size={50} />
    </Link>
  </footer>
)

export default Footer
