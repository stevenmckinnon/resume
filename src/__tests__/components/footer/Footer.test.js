import Enzyme, { shallow } from 'enzyme';
import React from 'react';
import Adapter from 'enzyme-adapter-react-16';
import Footer from '../../../components/footer/Footer';

Enzyme.configure({ adapter: new Adapter() });

describe('Tests the Footer component', () => {
    it('renders the Footer component', () => {
        const name = 'Steve McKinnon';
        const socialMedia = {
            twitter: 'http://twitter.com/stevenmckinnon',
            linkedIn: 'https://www.linkedin.com/in/stevenmckinnon92/',
            instagram: 'http://instagram.com/stevenmckinnon',
            photography: 'http://stevenmckinnon.co.uk',
            github: 'https://github.com/stevenmckinnon'
        };
        const wrapper = shallow(<Footer name={name} socialMedia={socialMedia} />);
        expect(wrapper).toMatchSnapshot();
    });
    it('renders the Footer component', () => {
        const name = 'Steve McKinnon';
        const socialMedia = {};
        const wrapper = shallow(<Footer name={name} socialMedia={socialMedia} />);
        expect(wrapper).toMatchSnapshot();
    });
});
