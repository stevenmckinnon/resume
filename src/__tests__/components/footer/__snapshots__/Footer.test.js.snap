// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests the Footer component renders the Footer component 1`] = `ShallowWrapper {}`;

exports[`Tests the Footer component renders the Footer component 2`] = `ShallowWrapper {}`;
